tools/bowtie2.wdl

```diff
@@ -0,0 +1,447 @@
+        }
+        repetitive_seeds: {
+            description: "For reads with repetitive seeds, try N sets of seeds",
+            help: "N is the maximum number of times Bowtie 2 will re-seed reads with repetitive seeds. When re-seeding, Bowtie 2 simply chooses a new set of reads (same length, same number of mismatches allowed) at different offsets and searches for more alignments. A read is considered to have repetitive seeds if the total number of seed hits divided by the number of seeds that aligned at least once is greater than 300.",
+        }
+        n_ceil: {
+            description: "func for maximum number of non-A/C/G/Ts permitted in alignment",
+            help: "Sets a function governing the maximum number of ambiguous characters (usually Ns and/or .s) allowed in a read as a function of read length. For instance, specifying -L,0,0.15 sets the N-ceiling function f to f(x) = 0 + 0.15 * x, where x is the read length. Reads exceeding this ceiling are filtered out."
+        }
+    }
+
+    input {
+        File bowtie_db_tar_gz
+        File read_one_fastq_gz
+        ReadGroup read_group
+        String prefix
+        File? read_two_fastq_gz
+        String? addl_rg_text
+        String? sam_opt_config
+        Int? skip
+        Int? upto
+        Int? match_bonus
+        Int? max_aln_report
+        Bowtie2Function score_min = Bowtie2Function {
+            function_type: "L",
+            constant: -0.6,
+            coefficient: -0.6,
+        }
+        Bowtie2Function interval_seed_substrings = Bowtie2Function {
+            function_type: "S",
+            constant: 1,
+            coefficient: 1.15,
+        }
+        Bowtie2Function n_ceil = Bowtie2Function {
+            function_type: "L",
+            constant: 0,
+            coefficient: 0.15,
+        }
+        Pair[Int, Int] read_gap_open_extend = (5, 3)
+        Pair[Int, Int] ref_gap_open_extend = (5, 3)
+        Pair[Int, Int] mismatch_penalty = (6, 2)
+        Boolean end_to_end = true  # false: --local
+        Boolean non_deterministic = false
+        Boolean ignore_quals = false
+        Boolean nofw = false
+        Boolean norc = false
+        Boolean no_1mm_upfront = false
+        Boolean report_all_alignments = false
+        Boolean no_mixed = false
+        Boolean no_discordant = false
+        Boolean dovetail = false
+        Boolean no_contain = false
+        Boolean no_overlap = false
+        Boolean write_unpaired_unaligned = false
+        Boolean write_unpaired_aligned = false
+        Boolean write_paired_discordant = false
+        Boolean write_paired_concordant = false
+        Boolean metrics_file = false
```

Comment:
Should this be `true`? Gut instinct is that metrics reporting is sane to default into

Reply:
I don't think these metrics are useful. I'd have to run it with metrics enabled, but the internet indicates that these are internal counters for the algorithm, rather than anything too useful to a user. The manual only specifies that they have some utility for debugging.

> Write bowtie2 metrics to file <path>. Having alignment metric can be useful for debugging certain problems, especially performance issues. See also: --met. Default: metrics disabled.

Reply:
Ok. `false` is a fine default in that case.